cmd/launcher/launcher.go

```diff
@@ -99,7 +99,7 @@ func runLauncher(ctx context.Context, cancel func(), opts *launcher.Options) err
 	internal.RecordLauncherVersion(rootDirectory)
 
 	// Try to ensure useful info in the logs
-	checkpoint.Run(logger, db)
+	checkpoint.Run(logger, db, *opts)
```

Comment:
I passed the value instead of a pointer because I didn't think the checkpoint.Run func should have access to edit the options used by the rest of the application. I think the only argument for passing a pointer would be to save some memory? So I guess the questions is ... is the memory we save using a pointer worth giving unnecessary access to app wide settings? .... there is also the possibility that I could be completely missing something =)

Reply:
I guess another question could be ... is there ever a case where another part of the application edits the options and we would expect that to propagate to the checkpoint log? If so, we would need a pointer.